app/src/main/java/org/xbmc/kore/ui/sections/favourites/FavouritesListFragment.java

```diff
@@ -0,0 +1,224 @@
+                setupFavouritesList(result.items);
+                swipeRefreshLayout.setRefreshing(false);
+            }
+
+            @Override
+            public void onError(int errorCode, String description) {
+                if (!isAdded()) return;
+                LogUtils.LOGD(TAG, "Error getting favourites: " + description);
+
+                emptyView.setText(getString(R.string.error_favourites, description));
+                Toast.makeText(getActivity(), getString(R.string.error_favourites, description),
+                        Toast.LENGTH_SHORT).show();
+                swipeRefreshLayout.setRefreshing(false);
+            }
+        }, callbackHandler);
+    }
+
+    /**
+     * Called to set the GridView with the favourites that are coming from the host.
+     *
+     * @param favourites the favourites list that is supplied to the GridView.
+     */
+    private void setupFavouritesList(List<FavouriteType.DetailsFavourite> favourites) {
+        if (favouritesAdapter == null) {
+            favouritesAdapter = new FavouritesAdapter(getActivity(), R.layout.grid_item_channel,
+                    hostManager);
+        }
+        gridView.setAdapter(favouritesAdapter);
+
+        favouritesAdapter.clear();
+        favouritesAdapter.addAll(favourites);
+        favouritesAdapter.notifyDataSetChanged();
+    }
+
+    private static class FavouritesAdapter extends ArrayAdapter<FavouriteType.DetailsFavourite> {
+
+        private final HostManager hostManager;
+        private final int artWidth, artHeight;
+
+        FavouritesAdapter(@NonNull Context context, @LayoutRes int resource, HostManager hostManager) {
+            super(context, resource);
+            this.hostManager = hostManager;
+            Resources resources = context.getResources();
+            artWidth = (int) (resources.getDimension(R.dimen.channellist_art_width) /
+                    UIUtils.IMAGE_RESIZE_FACTOR);
+            artHeight = (int) (resources.getDimension(R.dimen.channellist_art_heigth) /
+                    UIUtils.IMAGE_RESIZE_FACTOR);
+        }
+
+        @NonNull
+        @Override
+        public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
+            if (convertView == null) {
+                convertView = LayoutInflater.from(getContext()).inflate(R.layout.grid_item_channel,
+                        parent, false);
+                final FavouriteItemViewHolder vh = new FavouriteItemViewHolder(convertView);
+                convertView.setTag(vh);
+            }
+
+            final FavouriteItemViewHolder vh = (FavouriteItemViewHolder) convertView.getTag();
+            final FavouriteType.DetailsFavourite favouriteDetail = getItem(position);
+
+            // We don't need the context menu here.
+            vh.contextMenu.setVisibility(View.GONE);
+
+            vh.titleView.setText(favouriteDetail.title);
+            vh.detailView.setText(favouriteDetail.type);
```

Comment:
Ideally the `favoriteDetail.type` shouldn't be used directly as the text. We should have the possible values on `strings.xml` and use those, so that they could be translated/made user friendly.